controllers/video.go

```diff
@@ -240,50 +260,92 @@ func GetManifestFile(w http.ResponseWriter, r *http.Request, db *sql.DB) {
 	client := &http.Client{}
 
 	response, err := client.Do(request)
-
 	if err != nil {
 		log.Println(err)
-		http.Error(w, "Error retrieving video", http.StatusInternalServerError)
+		http.Error(w, "Error Streaming Video", http.StatusInternalServerError)
 		return
 	}
-
 	defer response.Body.Close()
 
 	if response.StatusCode == 404 {
-		http.Error(w, "Video record not found", http.StatusNotFound)
+		http.Error(w, "Video chunk not found", http.StatusNotFound)
 		return
 	}
 
 	bodyBytes, err := io.ReadAll(response.Body)
 
 	if err != nil {
 		log.Println(err)
-		http.Error(w, "Error retrieving video", http.StatusInternalServerError)
+		http.Error(w, "Error Streaming Video", http.StatusInternalServerError)
 		return
 	}
 
-	w.Header().Set("Content-Type", "application/x-mpegURL")
+	w.Header().Set("Content-Type", "video/MP2T")
 	w.WriteHeader(http.StatusOK)
 	w.Write(bodyBytes)
 }
 
-func GetTSFiles(w http.ResponseWriter, r *http.Request, db *sql.DB) {
-	videoName := strings.Split(r.URL.Path[1:], "/")[3]
+func DeleteVideo(w http.ResponseWriter, r *http.Request, db *sql.DB) {
+	body, err := io.ReadAll(r.Body)
 
-	videoComps := strings.Split(videoName, ".")
-	hashChecksum := sha1.New()
-	hashChecksum.Write([]byte(videoComps[0]))
-	fileId := fmt.Sprintf("%x", hashChecksum.Sum(nil))[:36]
+	if err != nil {
+		log.Println(err)
+		http.Error(w, "Error Deleting Video", http.StatusInternalServerError)
+		return
+	}
 
-	log.Println("Video chunk requested: " + fileId)
+	videoId := string(body)
```

Comment:
If the video file is too big, there will be too many `.ts` files to delete.
This will make the person wait for too long.

We can refactor this logic where if the DELETE API is called, then it sets an attribute, for instance `to_delete` or `delete` as true.

and we add a `delete_video` job in the queue which takes care of the actual deletion process.

And in our list files query, we add a WHERE clause to exclude records which are marked to be deleted.

Reply:
After this PR is done, we can also think about a bulk deletion API, but for now, this one should do.